### DISCUSSION TITLE
Runtime Error on logout and giberrish route

### DISCUSSION THREAD

**Discussion by dvnrsn** [blitz-js/blitz#2822] [Q&A]
I'm observing some pages initiate render and custom hooks which are triggering some requests that are giving

`message: "You must be logged in to access this"`

The desire for these these pages is to not even begin render but to redirect to login.

I observe `<ErrorBoundary` wrapping everything which redirects to login on auth error so I'm confused why there is a runtime error? How can I prevent these runtime errors?

Edit: what seems to be the case is that there is a dynamic route [id] that is accepting anything and attempting to render but fails because user is not authenticated.

I've tried putting `Page.authenticate = true` on both the rendered component and the layout (it uses `.getLayout`.

I've also tried using `ctx.session.$authorize()` to stop the flow but the app doesn't recognize any session it seems?

**Comment by dvnrsn** [comment]
Trying `ctx.session.$authorize()` without success.

`TypeError: Cannot read property '$authorize' of undefined`

I'm wondering if the dynamic routing choice (immediately accepting ID as first in path) is the wrong approach

**Reply by beerose** [reply]
Hey @dvnrsn, I'm sorry for the late reply. 
I tried reproducing with dynamic routes, the dynamic parameter being first in the path, and other things you mentioned, but it seems to work fine. Could you convert it to an issue and provide more details about the environment etc.?

**Reply by flybayer** [reply]
Yeah, if you can share code/repo, it would really help to understand what's happening.

**Reply by dvnrsn** [reply]
I see now how to somewhat recreate with a new blitz app but this is most likely a problem on my end because a fresh project handles it gracefully.

1. Create file `pages/[someId]/index.tsx`
2. Export a component that makes use of `const user = useCurrentUser()`
3. In useCurrentUser add `session.$authorize()` near the top
4. Log out
5. Navigate to `locahost:3000/giberish`
6. Observe `AuthenticationError` in console but the login page (correctly) in the viewport

In my case I'm seeing the error is not caught
<img width="1150" alt="image" src="https://user-images.githubusercontent.com/9112811/138530282-d7907b4d-b4fb-4c50-99c5-a2f6f05dce51.png">

**Reply by dvnrsn** [reply]
The `RootErrorFallback` is being invoked correctly in my context. I'm not sure what more to do other than upgrade all packages which will happen in the next month.

Potentially relevant packages:

```js
"blitz": "0.32.1",
"react": "0.0.0-experimental-7f28234f8",
"react-dom": "0.0.0-experimental-7f28234f8",
"react-error-boundary": "3.1.1",
```

**Reply by dvnrsn** [reply]
What doesn't make any sense to me is if I remove `<ErrorBoundary` from the generic app's `_app.tsx`, the runtime exception is still caught? The screen is blank (as opposed to mine above) and the error shows in the console. 

<img width="516" alt="image" src="https://user-images.githubusercontent.com/9112811/138531789-10d2158b-7368-47bb-a5cc-2f4832e31ba5.png">

**Reply by flybayer** [reply]
Oh you are on an ancient Blitz version 😅 I highly recommend upgrading to the latest (just make sure you follow breaking change instructions for each release!)

**Reply by dvnrsn** [reply]
Thanks Brandon. Appreciate you two and all your work.
